src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/RoslynCompletionSet.cs

```diff
@@ -63,6 +62,10 @@ public override void SelectBestMatch()
         {
             // Do nothing.  We do *not* want the default behavior that the editor has.  We've
             // already computed the best match.
+
+            // this will get called right after completion set is painted on the screen.
```

Comment:
I think we just told the editor we're not using this, and this seems a pretty hackish side effect for this method. What's the right place for it?

Reply:
there is no right place. if you are talking with editor team on changing intellisense API, we should make sure we can call start session when we really start session rather than the hacky way we are doing with intellisense API, and move this telemetry completely to them.

Reply:
@AmadeusW for this feedback then. Have they signed off on this hack?

Reply:
I believe he is the one who suggested the hack.